### PR TITLE
doc(bigtable): create page for configuration options

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10197)

### CODE REVIEW

File: google/cloud/bigtable/options.h [hunk 1]
```diff
@@ -145,17 +151,29 @@ using DataLimitedErrorCountRetryPolicy =
     ::google::cloud::internal::LimitedErrorCountRetryPolicy<
         bigtable::internal::SafeGrpcRetry>;
 
-/// Option to configure the retry policy used by `Table`.
+/**
+ * Option to configure the retry policy used by `Table`.
+ *
+ * @ingroup bigtable-options
+ */
 struct DataRetryPolicyOption {
```
Comment (dbolduc) [inline google/cloud/bigtable/options.h:159]: every generated service has an equivalent of this option too. Will they all get these `*-options.dox`?

**Review by dbolduc — COMMENTED** [review]
Should we be adding the `OptionList<>`s? Have they fallen out of favor?

**Comment by coryan** [comment]
> Should we be adding the `OptionList<>`s? Have they fallen out of favor?

I am not sure why customers would care for them.  AFAIK they are only used to implement some checks, and probably should not be in the public namespace.  I have no idea if we should remove them, move them to `*_internal`.  Until we decide, we should leave them undocumented (created #10206) to track.

File: google/cloud/bigtable/options.h (hunk 1 above)
Comment (coryan) [reply google/cloud/bigtable/options.h:159]: I would like to, yes, but I did the hand-crafted libraries first.

**Review by dbolduc — APPROVED** [review]
Ack. I think the `OptionList<>`s were also added for discoverability of Options. https://github.com/googleapis/google-cloud-cpp/blob/3baceacc1fde7f36d478df153cc113b857798335/google/cloud/accessapproval/access_approval_connection.h#L118-L125

But these doxygen ref groups probably do a better job.

**Pull request merged**